src/engine/Algebra.ts

```diff
@@ -1,10 +1,13 @@
+/// <reference path="Interfaces/ConstructorBags/AlgebraOptions.ts" />
+
 module ex {
 
    /**
     * A 2D vector on a plane. 
     */
-   export class Vector {
-
+   export class Vector implements IVectorOptions {
```

Comment:
The `implements IVectorOptions` seems unnecessary.  You can already pass a `Vector` to the constructor because it implicitly implements the `IVectorOptions` interface.